unitary/alpha/quokka_sampler.py

```diff
@@ -0,0 +1,161 @@
+
+    See https://www.quokkacomputing.com/ for more information.a
+
+    Args:
+        name: name of your quokka device
+        endpoint: HTTP url endpoint to post queries to.
+        post_function: used only for testing to override default
+            behavior to connect to internet URLs.
+    """
+
+    def __init__(
+        self,
+        name: str = _DEFAULT_QUOKKA_NAME,
+        endpoint: Optional[str] = None,
+        post_function: Optional[Callable[[JSON_TYPE], JSON_TYPE]] = None,
+    ):
+        self.quokka_name = name
+        self.endpoint = endpoint
+        self.post_function = post_function
+
+        if self.endpoint is None:
+            self.endpoint = _REQUEST_ENDPOINT.format(self.quokka_name)
+        if self.post_function is None:
+            self.post_function = self._post
+
+    def _post(self, json_request: JSON_TYPE) -> JSON_TYPE:
+        """Sends POST queries to quokka endpoint."""
+        try:
+            import requests
+        except ImportError as e:
+            print(
+                "Please install requests library to use Quokka"
+                "(e.g. pip install requests)"
+            )
+            raise e
+        result = requests.post(self.endpoint, json=json_request)
+        return json.loads(result.content)
+
+    def run_sweep(
+        self,
+        program: "cirq.AbstractCircuit",
+        params: "cirq.Sweepable",
+        repetitions: int = 1,
+    ) -> Sequence["cirq.Result"]:
+        """Samples from the given Circuit.
+
+        This allows for sweeping over different parameter values,
+        unlike the `run` method.  The `params` argument will provide a
+        mapping from `sympy.Symbol`s used within the circuit to a set of
+        values.  Unlike the `run` method, which specifies a single
+        mapping from symbol to value, this method allows a "sweep" of
+        values.  This allows a user to specify execution of a family of
+        related circuits efficiently.
+
+        Args:
+            program: The circuit to sample from.
+            params: Parameters to run with the program.
+            repetitions: The number of times to sample.
+
+        Returns:
+            Result list for this run; one for each possible parameter resolver.
+        """
+        rtn_results = []
+        qubits = sorted(program.all_qubits())
+        measure_keys = {}
+        register_names = {}
+        meas_i = 0
+
+        # Find all measurements in the circuit and record keys
+        # so that we can later translate between circuit and QASM registers.
+        for op in program.all_operations():
+            if isinstance(op.gate, cirq.MeasurementGate):
+                key = cirq.measurement_key_name(op)
+                if key in measure_keys:
+                    print(
+                        "Warning!  Keys can only be measured once in Quokka simulator"
+                    )
+                    print("Key {key} will only contain the last measured value")
```

Comment:
add `file=sys.stderr` to prints if these are warnings?

Reply:
Changed to warnings.warn.